massa-execution/src/vm.rs

```diff
@@ -311,6 +311,7 @@ impl VM {
             slot: step.slot,
```

Comment:
I comment here because Github doesn't let me comment at the line I want.

* clear_and_update_context should include a reset to zero of the event counter as well as a clear() of the list of generated events

* prepare_context should also return a cloned snapshot of the events that can be rollbacked in case of failure, just like `context.ledger_step.caused_changes = ledger_changes_backup;` for example